vscode-trace-extension/src/utils/backend-tsp-client-provider.ts

```diff
@@ -9,32 +9,36 @@ import { TspClientProvider } from 'vscode-trace-common/lib/client/tsp-client-pro
  *  for the Trace Extension's VSCode Backend
  */
 
-let _root: string;
-let _path: string;
-let _url: string;
+let _rootBE: string;
+let _rootFE: string;
+let _urlBE: string;
+let _urlFE: string;
 let _provider: TspClientProvider;
 
-export const getTraceServerUrl = (): string => _root;
-export const getTspApiEndpoint = (): string => _path;
-export const getTspClientUrl = (): string => _url;
+export const getTraceServerUrl = (isFrontEnd: boolean): string => (isFrontEnd ? _rootFE : _rootBE);
+export const getTspClientUrl = (isFrontEnd: boolean): string => (isFrontEnd ? _urlFE : _urlBE);
 
 export const getTspClient = (): TspClient => _provider.getTspClient();
 export const getExperimentManager = (): ExperimentManager => _provider.getExperimentManager();
 export const getTraceManager = (): TraceManager => _provider.getTraceManager();
 
 export const updateTspClientUrl = async (): Promise<string> => {
-    const _extUri = await getExternalUriFromUserSettings();
-    _root = _extUri.toString();
-    _path = getApiPathFromUserSettings();
-    _url = _root + _path;
+    const extUri = await getExternalUriFromUserSettings(false);
+    _rootBE = extUri.toString();
+    const apiPath = getApiPathFromUserSettings();
+    _urlBE = _rootBE + apiPath;
+
+    const extUriFE = await getExternalUriFromUserSettings(true);
+    _rootFE = extUriFE.toString();
+    _urlFE = _rootFE + apiPath;
 
     if (!_provider) {
-        _provider = new TspClientProvider(_url, undefined);
+        _provider = new TspClientProvider(_urlBE, undefined);
     } else {
-        _provider.updateTspClientUrl(_url);
+        _provider.updateTspClientUrl(_urlBE);
     }
 
-    return _url;
+    return _urlBE;
```

Comment:
Shouldn't `_urlFE` be returned here. This is called to update the front-end. Maybe to make it clearer is to not have this method return anything, but call getTraceServerUrl(true) to get the URL to propagate. See extension.ts code above where it is propagated when the URL is changed.

Reply:
Opted to remove the return value and use ```etTspClientUrl(true);```
If you agree please resolve the conversation, or let me know if I should do adjustments.

The README has also been updated.